### PR TITLE
Culturocity

### PR DESCRIPTION
**Pull request by Zemurin**
.. where to even begin.

### CODE REVIEW

**Comment by Zemurin** [comment]
This fixes and implements:
#30 - dynamic generation of cultures.txt 
#32 - detection of native cultures
#33 - split unmapped colonial cultures automatically
#47 - configurable assimilation rates for superregions
#92 - ditto
- parsing of entire eu4 localization so it can be used for anything, but currently only for neocultures
- primary and accepted cultures remapped to neocultures where sensible

other stuff:
#388 - detection of eu4 install on linux
- broken Liege mapping

**Comment by Zemurin** [comment]
This PR is now frozen, there should be no further commits to it, though most of the new stuff dealt with metadata and tests.

File: EU4toV2/Source/Mappers/CultureGroups/CultureGroups.cpp [hunk 1]
```diff
@@ -40,35 +50,75 @@ void mappers::CultureGroups::registerKeys()
...
-				const auto& oldCultures = itr->second;
-				for (const auto& oldCulture: oldCultures)
+				// We would normally override base definitions with incoming CK2 ones, but CK2 definitions
+				// are crap and don't actually list all required cultures, so we have to merge.
+				for (const auto& cultureItr: newGroup.getCultures())
 				{
-					cultures.push_back(oldCulture);
-				}
+					cultureGroupsMap[cultureGroupName].mergeCulture(cultureItr.first, cultureItr.second);
+				}
```
Comment (Idhrendur) [inline EU4toV2/Source/Mappers/CultureGroups/CultureGroups.cpp]: nit: whitespace randomly inserted at the end of line 60 here.

File: EU4toV2/Source/Mappers/CultureMapper/CultureMapper.h [hunk 2]
```diff
@@ -25,10 +29,24 @@ namespace mappers
...
+			const std::string& eu4culture,
+			const std::string& eu4religion,
+			int eu4Province,
+			const std::string& eu4ownerTag) const;
+
 	private:
 		void registerKeys();
 		
-		std::vector<CultureMappingRule> cultureMapRules;
+		std::list<CultureMappingRule> cultureMapRules;
```
Comment (Idhrendur) [inline EU4toV2/Source/Mappers/CultureMapper/CultureMapper.h]: Due to caching, it seems as if vector is actually faster: https://baptiste-wicht.com/posts/2012/12/cpp-benchmark-vector-list-deque.html

Probably best to keep it (unless you saw performance differences favoring list, in which case disregard).

File: EU4toV2/Source/Mappers/CultureGroups/CultureGroups.cpp (hunk 1 above)
Comment (Zemurin) [reply EU4toV2/Source/Mappers/CultureGroups/CultureGroups.cpp]: dropped.

File: EU4toV2/Source/Mappers/CultureMapper/CultureMapper.h (hunk 2 above)
Comment (Zemurin) [reply EU4toV2/Source/Mappers/CultureMapper/CultureMapper.h]: yeah, was about to shuffle rules internally according to distinguishers but it turned out unnecessary. fixed

**Pull request merged**